### PR TITLE
Fix portrait endpoint for email address user id's

### PR DESCRIPTION
**Pull request by instification**
The previous pr #1466 resolved an issue where the endpoint was returning unsanitized user id's.

Following #1481 the new @portrait endpoint does not require sanitized user id's so when email login is enabled, portraits break.

The solution would either be to revert the change made in #1466 and return the raw user id, or to update the @portrait endpoint to handle sanitized id's.

This PR does the latter as well as:

 - Updating the endpoint for discussion/comments to use the new url
 - Updates the tests
   - Uses a Plone user instead of the zmi admin user, as this causes issues rendering the portrait
   - Adds a tests for users where login by email is enabled
   - Updates comments tests

### CODE REVIEW

**Comment by instification** [comment]
@jenkins-plone-org please run jobs

**Review by tisto — APPROVED** [review]
LGTM. @instification thanks for the fix!

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/services/users/get.py
Comment (jensens) [inline src/plone/restapi/services/users/get.py:225]: Here be dragons.

There is a long (>15y) history of patching `MembershipTool.getPersonalPortrait` in several addons dealing with it. Membrane did/does, pas.plugins.ldap as well, probably others too.

Before you merge this please check if this works with a recent version of `pas.plugins.ldap` or similar (should, but better safe than sorry).  I am also not sure if the meta_type assumption  always holds.

Comment (instification) [reply src/plone/restapi/services/users/get.py:225]: So if I understand your comment @jensens you're suggesting it would be dangerous to not use `getPersonalPortrait`?

Because other addon's may patch it to give a different result, so by not using it we could be missing out on those changes?

I can see in pas.plugins.ldap it is patching it to basically solve the issue we are seeing here: https://github.com/collective/pas.plugins.ldap/blob/e1f8da6ab300bed50c779febac9036930b2cb91f/src/pas/plugins/ldap/monkey.py#L58-L86

Unfortunately, `portal_membership.getPersonalPortrait` doesn't return a portrait if the user id is already sanitised.

Perhaps the better solution would be to 'de-sanitize' the user id before passing it to `getPersonalPortrait`?

Comment (jensens) [reply src/plone/restapi/services/users/get.py:225]: Exact, yes. And I am not sure if the meta_type is always the one you expect.

Comment (instification) [reply src/plone/restapi/services/users/get.py:225]: ok thanks for the feedback. I'll take another look at the meta_type check, I'm sure there is a better way to determine if the users portrait is unset.

Comment (instification) [reply src/plone/restapi/services/users/get.py:225]: @jensens I have refactored the code.

So now both the code that returns the portrait url and fetches the portrait will use getPersonalPortrait.

When fetching from the @portrait endpoint, it is expecting a 'safe id' so is calling `decleanId` from pas before calling `getPersonalPortrait`.

I don't really have an effective setup to test pas.ldap (and certainly not with custom portrait code), but as we are using the `getPersonalPortrait` function as intended I can't see it being an issue there due to this change.

**Comment by instification** [comment]
@sneridagh I have also updated the endpoints/tests for comments as they were returning the old portrait url, which was causing it to break in volto. I can refactor the change into a new PR if you think it is better to do it that way

**Comment by instification** [comment]
@jenkins-plone-org please run jobs

**Comment by instification** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/services/users/get.py
Comment (jensens) [reply src/plone/restapi/services/users/get.py:225]: Yes, this will do it. Great work. It is always good to see others getting their fingers dirty with our membership code ;-)

**Review by jensens — APPROVED** [review]

**Comment by instification** [comment]
I noticed in local testing that the `isDefaultPortrait` function was always returning true, even when a users portrait wasn't set.

For some reason `getattr(portal, default_portrait, None)` always returns None, but `portal.unrestrictedTraverse(default_portrait, None)` does return the root portrait. I guess at some point in the past this was changed from being an object in zope to something else.

Additionally, even though `portal_membership.getPersonalPortrait('admin`)` returns `<FSImage /Plone/defaultUser.png>` and the traversed default portrait is also `<FSImage /Plone/defaultUser.png>` they don't compare. So I updated the function to compare on `.getPhysicalPath()` which does match for some reason.

I added a test to check for a 404 from the endpoint when the portrait isn't set.

**Comment by instification** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/test_services_users.py
Comment (davisagli) [inline src/plone/restapi/tests/test_services_users.py:1047]:
```suggestion
    def test_get_user_default_portrait(self):
```

**Pull request merged**
